signer/src/storage/model.rs

```diff
@@ -1352,6 +1349,260 @@ pub struct WithdrawalRejectEvent {
+        Self(value as u64)
+    }
+}
+impl From<u32> for StacksBlockHeight {
+    fn from(value: u32) -> Self {
+        Self(value as u64)
+    }
+}
+impl From<u64> for StacksBlockHeight {
+    fn from(value: u64) -> Self {
+        Self(value)
+    }
+}
+impl From<usize> for StacksBlockHeight {
+    fn from(value: usize) -> Self {
+        Self(value as u64)
+    }
+}
+
+// Conversion u64 => StacksBlockHeight is not implemented intentionally.
+// Use deref instead.
+
+impl From<StacksBlockHeight> for u128 {
+    fn from(value: StacksBlockHeight) -> Self {
+        *value as u128
+    }
+}
+
+impl std::fmt::Display for StacksBlockHeight {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+impl Deref for StacksBlockHeight {
+    type Target = u64;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+impl TryFrom<StacksBlockHeight> for i64 {
+    type Error = TryFromIntError;
+    fn try_from(value: StacksBlockHeight) -> Result<Self, Self::Error> {
+        i64::try_from(value.0)
+    }
+}
+
+impl TryFrom<i64> for StacksBlockHeight {
+    type Error = TryFromIntError;
+    fn try_from(value: i64) -> Result<Self, Self::Error> {
+        Ok(Self(<u64>::try_from(value)?))
+    }
+}
+
+impl Add<u64> for StacksBlockHeight {
+    type Output = StacksBlockHeight;
+    fn add(self, other: u64) -> Self::Output {
+        Self(self.0.add(other))
+    }
+}
+impl Add<StacksBlockHeight> for u64 {
+    type Output = StacksBlockHeight;
+    fn add(self, other: StacksBlockHeight) -> Self::Output {
+        StacksBlockHeight((self).add(other.0))
+    }
+}
+impl Add for StacksBlockHeight {
+    type Output = StacksBlockHeight;
+    fn add(self, other: StacksBlockHeight) -> Self::Output {
+        Self(self.0.add(other.0))
+    }
+}
+
+// I don't see any case where int - height can make sense.
+
+impl Sub<u64> for StacksBlockHeight {
+    // Height - int is still height.
+    type Output = StacksBlockHeight;
+    fn sub(self, other: u64) -> Self::Output {
+        StacksBlockHeight((*self).sub(other))
+    }
+}
+impl Sub for StacksBlockHeight {
+    // Diff of two heights is int, not height.
+    type Output = u64;
+    fn sub(self, other: StacksBlockHeight) -> Self::Output {
+        self.0.sub(other.0)
+    }
+}
+impl StacksBlockHeight {
+    /// Behaves same as u64.saturating_add
+    pub fn saturating_add(self, rhs: impl Into<StacksBlockHeight>) -> Self {
+        let rhs: u64 = rhs.into().0;
+        Self(self.0.saturating_add(rhs))
+    }
+
+    /// Behaves same as u64.saturating_sub
+    pub fn saturating_sub(self, rhs: impl Into<StacksBlockHeight>) -> Self {
+        let rhs: u64 = rhs.into().0;
+        Self(self.0.saturating_sub(rhs))
+    }
+}
+
+/// Bitcoin block height
+#[derive(
+    Debug,
+    Default,
+    Clone,
+    Copy,
+    Hash,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Serialize,
+    Deserialize,
+    sqlx::FromRow,
+)]
+#[sqlx(transparent)]
```

Comment:
nit: it doesn't seem we are using `sqlx(transparent)` (since we implemented sqlx encode/decode), do we need this?

Reply:
[e34aa64](https://github.com/stacks-sbtc/sbtc/pull/1448/commits/e34aa640c7011ca3e285c4a5a27c8d053ea71e46)